src/proto/generation.proto

```diff
@@ -240,6 +241,92 @@ message ClassifierParameters {
   optional Action realized_action = 3;
 }
 
+//
+// Transforms
+//
+
+enum BorderMode {
+  BORDER_REFLECT = 0;   // reflect image values across the border
+  BORDER_REPLICATE = 1; // replicate border values outside the image
+  BORDER_WRAP = 2;      // wrap around / tile the image values
+  BORDER_ZERO = 3;      // use 0 for locations outside the image
+}
+
+enum ColorMatchMode {
+  COLOR_MATCH_HSV = 0;  // match hue, saturation, and value histograms
+  COLOR_MATCH_LAB = 1;  // match lightness, a, and b histograms
+  COLOR_MATCH_RGB = 2;  // match red, green, and blue histograms
+}
+
+message TransformAddNoise {
+  float amount = 1; // amount of noise to add
```

Comment:
what's the range?

Reply:
it's for adding in random values in normal distribution (mean 0, variance 1). so amount 0 would be no additional noising, 1 would be full addition of gaussian noise. generally useful at low values like 0.02. "amount of gaussian noise to add" could be better comment. there isn't really a set range, not sure if you have idea how to better document it?

Reply:
This is a comment saying, 'this needs to be documented in the code, in a comment, what the range is'.